build/ci/phase-template.yml

```diff
@@ -36,5 +38,23 @@ phases:
         testRunTitle: Machinelearning_Tests_$(_phaseName)_$(_configuration)_$(Build.BuildNumber)
         configuration: $(_configuration)
         mergeTestResults: true
+    - task: CopyFiles@2
+      displayName: Stage build logs
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)
+        contents: '?(msbuild.*|binclash.log|init-tools.log)'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+    - task: CopyFiles@2
+      displayName: Stage test output
+      inputs:
+        sourceFolder: $(Build.SourcesDirectory)/bin
+        contents: '**/TestOutput/**/*'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+    - task: PublishBuildArtifacts@1
```

Comment:
does it gets zipped by default?

Reply:
I'm not 100% sure what is happening behind the scenes, but I think it does get zipped by default.  (The published artifact downloads as a zip file, and it gives the option to use multithreading to speed it up.)